updater/cmd/updater/main.go

```diff
@@ -60,11 +74,35 @@ func main() {
 
 	if err := installer.Install(rb); err != nil {
 		log.Default().Printf("Failed to install: %s", err)
-		// TODO: Change packagestatuses to have error message
+
+		// Set the state to failed before rollback so collector knows it failed
+		if setErr := monitor.SetState(packagestate.DefaultFileName, protobufs.PackageStatus_InstallFailed, err); setErr != nil {
+			log.Println("Failed to set state on install failure:", setErr)
+		}
 		rb.Rollback()
 		log.Default().Fatalf("Rollback complete")
 	}
 
-	// TODO: Monitor post install packagestatuses for failure/success
-	// If it doesn't pass, rollback
+	// Create a context with timeout to wait for a success or failed status
+	checkCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Monitor the install state
+	if err := monitor.MonitorForSuccess(checkCtx, packagestate.DefaultFileName); err != nil {
+		log.Println("Failed to install:", err)
+
+		// If this is not an error due to the collector setting a failed status we need to set a failed status
+		if !errors.Is(err, state.ErrFailedStatus) {
+			// Set the state to failed before rollback so collector knows it failed
+			if setErr := monitor.SetState(packagestate.DefaultFileName, protobufs.PackageStatus_InstallFailed, err); setErr != nil {
```

Comment:
@cpheps again, would be great if we set the error message here as well

Reply:
Same as above it sets the err message because the passed in error is not nil.